dev-notes/minimal-os-for-fobnail.md

```diff
@@ -0,0 +1,718 @@
+system.
+
+#### Running in DLME
+
+Ensuring that the operating system works in DLME can be achieved by using the
+[TrenchBoot](https://trenchboot.org/) project. For this purpose, a Yocto
+[meta-fobnail](https://github.com/fobnail/meta-fobnail) layer has been created
+that integrates the necessary elements. The effects of the tests are presented
+in a [separate document](./running-os-in-dlme.md).
+
+#### Fobnail integration
+
+At this point, the [Fobnail Attester](https://github.com/fobnail/fobnail-attester)
+application is strongly dependent on Linux. Mainly due to the fact that its
+development took place on this operating system. Therefore, it is important that
+the created minimal OS also has an integrated attester application. The main
+dependencies of its operation are drivers for USB, USB EEM and TPM, but running
+them under Linux will not cause major problems.
+
+### Linux with Busybox userspace
+
+The main advantage of Busybox based userspace is its small size. Also booting
+time would be significantly faster than userspace with fully-featured init
+system. With Busybox, init system can be replaced by a simple shell script.
+
+#### Running in DLME
+
+Since we are still running Linux, there are no problems here.
+
+#### Fobnail integration
+
+Since busybox based userspace doesn't have big expectation of Linux feature set,
+kernel can be reduced. Busybox can be compiled with only those features we need.
+Currently, Fobnail Attester depends on another program to configure network
+interface created by Fobnail Token. This is `systemd-networkd`, but in Busybox
+userspace it would have to be something else. It should be noted that
+configuration should be done on hotplug event, simply using `ifconfig` may be
+unreliable because:
+
+* It won't work if Fobnail Token isn't already plugged in.
+* It won't work if Fobnail Token is plugged out and back in.
+
+Hotplug detection and interface setup could be implemented in
+[Fobnail Attester](https://github.com/fobnail/fobnail-attester) through Netlink.
+
+### Linux with Go userspace
+
+[u-root](https://github.com/u-root/u-root) provides an easy way to deploy Go
+programs as standalone initramfs images. It is used by the
+[LinuxBoot](https://www.linuxboot.org/) project for creating small Linux distro
+that acts as bootloader for other OSes (through kexec).
+
+#### Running in DLME
+
+Since we are still running Linux, there are no problems here.
+
+#### Fobnail integration
+
+[Fobnail Attester](https://github.com/fobnail/fobnail-attester) is written in C,
+and it rather doesn't make sense to integrate it with Go userland. Also, it
+should be noted that `u-root` built initramfs contains only executable - all Go
+modules are linked into a single binary. Unless we want to rewrite Fobnail
+Attester there is no point in using userspace that was designed be Go-only.
+Rewriting Fobnail Attester in Go may give benefits of producing less vurnelable
+code.
```

Comment:
Wouldn't it be possible to Fobnail Attester as init? Do we need anything else from userspace? I think `kexec` can be run directly from attester.

@krystian-hebel ?